Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptAddController.java

```diff
@@ -51,117 +55,129 @@
+     *            Holds http request object information
+     * @param principal
+     *            holds log in information
+     * @return returns string which redirects to concept list page
+     * @throws LuceneException
+     * @throws DictionaryModifyException
+     * @throws DictionaryDoesNotExistException
+     * @throws IllegalAccessException
+     * @throws IndexerRunningException
+     */
+    @RequestMapping(value = "auth/conceptlist/addconcept/add", method = RequestMethod.POST)
+    public String addConcept(HttpServletRequest req, Principal principal,
+            @Validated @ModelAttribute("conceptAddBean") ConceptAddBean conceptAddBean, ModelMap model, BindingResult result)
+                    throws LuceneException, DictionaryDoesNotExistException, DictionaryModifyException,
+                    IllegalAccessException, IndexerRunningException {
+        if (result.hasErrors()) {
+            return "/auth/conceptlist/addconcept";
+        }
         ConceptEntry conceptEntry = new ConceptEntry();
-        conceptEntry.setSynonymIds(req.getParameter("synonymsids"));
-        conceptEntry.setWord(req.getParameter("name"));
-        conceptEntry.setConceptList(req.getParameter("lists"));
-        conceptEntry.setPos(req.getParameter("pos"));
-        conceptEntry.setDescription(req.getParameter("description"));
-        conceptEntry.setEqualTo(req.getParameter("equals"));
-        conceptEntry.setSimilarTo(req.getParameter("similar"));
-        conceptEntry.setTypeId(req.getParameter("types"));
-        conceptEntry.setCreatorId(principal.getName());
-        
-        //Checking if indexer is already running.
-        if(indexService.isIndexerRunning()){
-            model.addAttribute("show_error_alert", true);
-            model.addAttribute("error_alert_msg", indexerRunning);
-        	return "forward:/auth/conceptlist/addconcept";
+        try {
+            conceptEntry.setSynonymIds(conceptAddBean.getSynonymsids());
+            conceptEntry.setWord(conceptAddBean.getName());
+            conceptEntry.setConceptList(conceptAddBean.getSelectedList());
+            conceptEntry.setPos(conceptAddBean.getPos());
+            conceptEntry.setDescription(conceptAddBean.getDescription());
+            conceptEntry.setEqualTo(conceptAddBean.getEquals());
+            conceptEntry.setSimilarTo(conceptAddBean.getSimilar());
+            conceptEntry.setTypeId(conceptAddBean.getSelectedTypes());
+            conceptEntry.setCreatorId(principal.getName());
+
+            // Checking if indexer is already running.
+            if ((indexService.isIndexerRunning())) {
+                model.addAttribute("show_error_alert", true);
+                model.addAttribute("error_alert_msg", indexerRunning);
+                return "forward:/auth/conceptlist/addconcept";
+            }
+            conceptManager.addConceptListEntry(conceptEntry);
+
+        } catch (DictionaryDoesNotExistException e) {
+            e.printStackTrace();

```

Comment:
log the exception and show the user a flash message that says that the concept couldn't be added.